core/domain/feedback_validators_test.py

```diff
@@ -0,0 +1,746 @@
+                'Original author ID %s is in a wrong format. '
+                'It should be either pid_<32 chars> or uid_<32 chars>.\']]'
+            ) % (
+                self.model_instance.id, self.model_instance.original_author_id)
+        ]
+        self.run_job_and_check_output(
+            expected_output, sort=True, literal_eval=False)
+
+    def test_wrong_last_nonempty_message_author_id_format_failure(self):
+        self.model_instance.last_nonempty_message_author_id = 'wrong_id'
+        self.model_instance.update_timestamps()
+        self.model_instance.put()
+        expected_output = [
+            (
+                u'[u\'failed validation check for final author '
+                'check of GeneralFeedbackThreadModel\', [u\'Entity id %s: '
+                'Last non-empty message author ID %s is in a wrong format. '
+                'It should be either pid_<32 chars> or uid_<32 chars>.\']]'
+            ) % (
+                self.model_instance.id,
+                self.model_instance.last_nonempty_message_author_id
+            )
+        ]
+        self.run_job_and_check_output(
+            expected_output, sort=True, literal_eval=False)
+
+    def test_missing_message_model_failure(self):
+        feedback_models.GeneralFeedbackMessageModel.get_by_id(
+            '%s.0' % self.thread_id).delete()
+        expected_output = [
+            (
+                u'[u\'failed validation check for message_ids field '
+                'check of GeneralFeedbackThreadModel\', '
+                '[u"Entity id %s: based on field message_ids having value '
+                '%s.0, expected model GeneralFeedbackMessageModel with '
+                'id %s.0 but it doesn\'t exist"]]') % (
+                    self.model_instance.id, self.thread_id, self.thread_id)]
+        self.run_job_and_check_output(
+            expected_output, sort=True, literal_eval=False)
+
+    def test_invalid_has_suggestion(self):
+        self.model_instance.has_suggestion = False
+        self.model_instance.update_timestamps()
+        self.model_instance.put()
+        expected_output = [
+            (
+                u'[u\'failed validation check for has suggestion '
+                'check of GeneralFeedbackThreadModel\', [u\'Entity id %s: '
+                'has suggestion for entity is false but a suggestion exists '
+                'with id same as entity id\']]'
+            ) % self.model_instance.id]
+        self.run_job_and_check_output(
+            expected_output, sort=True, literal_eval=False)
+
+    def test_model_with_invalid_entity_type(self):
+        expected_output = [
+            (
+                u'[u\'failed validation check for entity type check '
+                'of GeneralFeedbackThreadModel\', [u\'Entity id %s: Entity '
+                'type exploration is not allowed\']]'
+            ) % self.model_instance.id]
+        with self.swap(
+            prod_validators, 'TARGET_TYPE_TO_TARGET_MODEL', {}):
```

Comment:
This will fix the tests
```suggestion
            feedback_validators, 'TARGET_TYPE_TO_TARGET_MODEL', {}):
```

Reply:
Thanks! :)